From: "Matteo F. Vescovi" <mfv@debian.org>
Date: Mon, 13 May 2019 11:38:22 +0200
Subject: Show_dialog_instead_of_crashing

---
 src/rcw.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/src/rcw.c b/src/rcw.c
index c659245..8e9bd2b 100644
--- a/src/rcw.c
+++ b/src/rcw.c
@@ -44,6 +44,7 @@
 #include <stdlib.h>
 
 #include "remmina.h"
+#include "remmina_main.h"
 #include "rcw.h"
 #include "remmina_file.h"
 #include "remmina_file_manager.h"
@@ -3775,6 +3776,7 @@ GtkWidget* rcw_open_from_file_full(RemminaFile* remminafile, GCallback disconnec
 	RemminaConnectionWindow* cnnwin;
 	GtkWidget* tab;
 	gint i;
+	const gchar *msg;
 
 	/* Create the RemminaConnectionObject */
 	cnnobj = g_new0(RemminaConnectionObject, 1);
@@ -3783,6 +3785,17 @@ GtkWidget* rcw_open_from_file_full(RemminaFile* remminafile, GCallback disconnec
 	/* Create the RemminaProtocolWidget */
 	cnnobj->proto = remmina_protocol_widget_new();
 	remmina_protocol_widget_setup((RemminaProtocolWidget*)cnnobj->proto, remminafile, cnnobj);
+        if (remmina_protocol_widget_has_error((RemminaProtocolWidget*)cnnobj->proto)) {
+		GtkWindow *wparent;
+		wparent = remmina_main_get_window();
+		msg = remmina_protocol_widget_get_error_message((RemminaProtocolWidget*)cnnobj->proto);
+		dialog = gtk_message_dialog_new (wparent,  GTK_DIALOG_DESTROY_WITH_PARENT,
+				GTK_MESSAGE_ERROR, GTK_BUTTONS_CLOSE, "%s", msg);
+		gtk_dialog_run(GTK_DIALOG (dialog));
+		gtk_widget_destroy(dialog);
+		/* We should destroy cnnobj->proto and cnnobj now... ToDo: fix this leak */
+		return NULL;
+	}
 	/* Set a name for the widget, for CSS selector */
 	gtk_widget_set_name(GTK_WIDGET(cnnobj->proto), "remmina-protocol-widget");
 
