From: Giovanni Panozzo <giovanni@panozzo.it>
Date: Thu, 10 Feb 2022 12:22:34 +0100
Subject: Fix_crash_when_closing_connection

---
 src/rcw.c                     | 1 +
 src/remmina_protocol_widget.c | 3 +++
 2 files changed, 4 insertions(+)

diff --git a/src/rcw.c b/src/rcw.c
index 0c3e591..b750a16 100644
--- a/src/rcw.c
+++ b/src/rcw.c
@@ -3399,6 +3399,7 @@ void rco_closewin(RemminaProtocolWidget *gp)
 	if (cnnobj) {
 		cnnobj->remmina_file = NULL;
 		g_free(cnnobj);
+		gp->cnnobj = NULL;
 	}
 
 	remmina_application_condexit(REMMINA_CONDEXIT_ONDISCONNECT);
diff --git a/src/remmina_protocol_widget.c b/src/remmina_protocol_widget.c
index 7a7fbc4..9a7099c 100644
--- a/src/remmina_protocol_widget.c
+++ b/src/remmina_protocol_widget.c
@@ -1519,6 +1519,9 @@ static gboolean remmina_protocol_widget_dialog_mt_setup(gpointer user_data)
 	RemminaMessagePanel *mp;
 	const gchar *s;
 
+	if (d->gp->cnnobj == NULL)
+		return FALSE;
+
 	mp = remmina_message_panel_new();
 
 	if (d->dtype == RPWDT_AUTH) {
