From: Giovanni Panozzo <giovanni@panozzo.it>
Date: Tue, 20 Feb 2018 23:04:59 +0100
Subject: Switch_stats_URL_to_https

---
 remmina/src/remmina_stats_sender.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/remmina/src/remmina_stats_sender.c b/remmina/src/remmina_stats_sender.c
index eaa34dd..3f922f8 100644
--- a/remmina/src/remmina_stats_sender.c
+++ b/remmina/src/remmina_stats_sender.c
@@ -54,7 +54,7 @@
 
 #define PERIODIC_UPLOAD_INTERVAL_SEC    2678400
 
-#define PERIODIC_UPLOAD_URL "http://www.remmina.org/stats/upload_stats.php"
+#define PERIODIC_UPLOAD_URL "https://www.remmina.org/stats/upload_stats.php"
 
 
 static gint periodic_check_source;
