Description: remove call to freerdp_get_last_error
Author: Luca Falavigna <dktrkranz@debian.org>

Index: remmina/remmina-plugins/rdp/rdp_plugin.c
===================================================================
--- remmina.orig/remmina-plugins/rdp/rdp_plugin.c
+++ remmina/remmina-plugins/rdp/rdp_plugin.c
@@ -883,9 +883,6 @@ static gboolean remmina_rdp_main(Remmina
 	{
 		if (!rfi->user_cancelled)
 		{
-			UINT32 e;
-		
-			e = freerdp_get_last_error(rfi->instance->context);
 			remmina_plugin_service->protocol_plugin_set_error(gp, _("Unable to connect to RDP server %s"), rfi->settings->ServerHostname);
 
 		}
