Description: Do not use deprecated g_thread_init function.
Author: Luca Falavigna <dktrkranz@debian.org>

Index: remmina/remmina/src/remmina.c
===================================================================
--- remmina.orig/remmina/src/remmina.c	2012-02-11 15:17:12.654820026 +0100
+++ remmina/remmina/src/remmina.c	2012-02-11 16:42:36.882977020 +0100
@@ -194,7 +194,7 @@
 	textdomain(GETTEXT_PACKAGE);
 
 #ifdef HAVE_PTHREAD
-	g_thread_init (NULL);
+	g_type_init ();
 	gdk_threads_init ();
 #endif
 
