Description: Add missing libraries to target_link_libraries
Author: Daniel T. Chen <crimsun@ubuntu.com>
Forwarded: https://github.com/FreeRDP/Remmina/issues/182

Index: remmina/remmina/CMakeLists.txt
===================================================================
--- remmina.orig/remmina/CMakeLists.txt	2013-06-23 14:11:21.648354867 +0200
+++ remmina/remmina/CMakeLists.txt	2013-06-23 14:27:38.780321818 +0200
@@ -91,7 +91,7 @@
 add_executable(remmina ${REMMINA_SRCS})
 
 include_directories(${GTK_INCLUDE_DIRS})
-target_link_libraries(remmina ${GTK_LIBRARIES})
+target_link_libraries(remmina ${GTK_LIBRARIES} gmodule-2.0 X11 gdk_pixbuf-2.0 gio-2.0 gobject-2.0 pthread)
 
 find_suggested_package(LIBSSH)
 if(LIBSSH_FOUND)
