Description: Add shebangs to external tool scripts
Author: Luca Falavigna <dktrkranz@debian.org>

Index: remmina/remmina/external_tools/launcher.sh
===================================================================
--- remmina.orig/remmina/external_tools/launcher.sh
+++ remmina/remmina/external_tools/launcher.sh
@@ -1,3 +1,5 @@
+#!/bin/sh
+
 ####################
 # Main Script
 ####################
Index: remmina/remmina/external_tools/pause.sh
===================================================================
--- remmina.orig/remmina/external_tools/pause.sh
+++ remmina/remmina/external_tools/pause.sh
@@ -1,3 +1,5 @@
+#!/bin/sh
+
 # Waits for user key press.
 pause ()
 {
Index: remmina/remmina/external_tools/remmina_filezilla_sftp.sh
===================================================================
--- remmina.orig/remmina/external_tools/remmina_filezilla_sftp.sh
+++ remmina/remmina/external_tools/remmina_filezilla_sftp.sh
@@ -1,3 +1,5 @@
+#!/bin/sh
+
 . $(dirname $0)/pause.sh
 
 filezilla sftp://$ssh_username:$password@$server 
Index: remmina/remmina/external_tools/remmina_filezilla_sftp_pki.sh
===================================================================
--- remmina.orig/remmina/external_tools/remmina_filezilla_sftp_pki.sh
+++ remmina/remmina/external_tools/remmina_filezilla_sftp_pki.sh
@@ -1,3 +1,5 @@
+#!/bin/sh
+
 . $(dirname $0)/pause.sh
 
 filezilla sftp://$ssh_username@$server 
Index: remmina/remmina/external_tools/remmina_nslookup.sh
===================================================================
--- remmina.orig/remmina/external_tools/remmina_nslookup.sh
+++ remmina/remmina/external_tools/remmina_nslookup.sh
@@ -1,3 +1,5 @@
+#!/bin/sh
+
 . $(dirname $0)/pause.sh
 
 nslookup $server
Index: remmina/remmina/external_tools/remmina_ping.sh
===================================================================
--- remmina.orig/remmina/external_tools/remmina_ping.sh
+++ remmina/remmina/external_tools/remmina_ping.sh
@@ -1,3 +1,5 @@
+#!/bin/sh
+
 . $(dirname $0)/pause.sh
 
 ping -c3 $server
Index: remmina/remmina/external_tools/remmina_traceroute.sh
===================================================================
--- remmina.orig/remmina/external_tools/remmina_traceroute.sh
+++ remmina/remmina/external_tools/remmina_traceroute.sh
@@ -1,3 +1,5 @@
+#!/bin/sh
+
 . $(dirname $0)/pause.sh
 
 traceroute $server
