Description: Mark main toolbar as a primary toolbar
Author: Robert Ancell <robert.ancell@canonical.com>
Bug: http://bugs.debian.org/659781

Index: remmina/remmina/src/remmina_main.c
===================================================================
--- remmina.orig/remmina/src/remmina_main.c	2012-02-23 22:15:39.953762296 +0100
+++ remmina/remmina/src/remmina_main.c	2012-02-23 22:40:59.633808853 +0100
@@ -1120,6 +1120,7 @@
 	gtk_box_pack_start(GTK_BOX(vbox), menubar, FALSE, FALSE, 0);
 
 	priv->toolbar = gtk_ui_manager_get_widget(uimanager, "/ToolBar");
+	gtk_style_context_add_class(gtk_widget_get_style_context(priv->toolbar), GTK_STYLE_CLASS_PRIMARY_TOOLBAR);
 	gtk_box_pack_start(GTK_BOX(vbox), priv->toolbar, FALSE, FALSE, 0);
 
 	remmina_main_create_quick_search(remminamain);
